src/DsbNorge.A3Forms/Clients/Brreg/BrregClient.cs

```diff
@@ -0,0 +1,208 @@
+using System.Globalization;
+using System.Net;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using DsbNorge.A3Forms.Models;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Logging;
+
+namespace DsbNorge.A3Forms.Clients.Brreg;
+
+public class BrregClient : IBrregClient
+{
+    private readonly JsonSerializerOptions _serializerOptions;
+    private readonly MemoryCacheEntryOptions _cacheOptions;
+    private readonly ILogger<IBrregClient> _logger;
+    private readonly IMemoryCache _memoryCache;
+    private readonly HttpClient _client;
+
+    public BrregClient(
+        HttpClient client,
+        ILogger<IBrregClient> logger,
+        IMemoryCache memoryCache
+        )
+    {
+        _logger = logger;
+        _client = client;
+        _client.BaseAddress = new Uri("https://data.brreg.no/");
+        _memoryCache = memoryCache;
+        _serializerOptions = new JsonSerializerOptions
+        {
+            PropertyNameCaseInsensitive = true
+        };
+        _cacheOptions = new MemoryCacheEntryOptions
+        {
+            AbsoluteExpirationRelativeToNow = TimeSpan.FromHours(24)
+        };
+    }
+    
+    public async Task<BrregOrg?> GetOrg(string orgNumber)
+    {
+        try
+        {
+            if (_memoryCache.TryGetValue(orgNumber, out BrregOrg? cachedOrg))
+            {
+                _logger.LogInformation($"Retrieved organization {orgNumber} from cache");
+                return cachedOrg;
+            }
+
+            var path = $"/enhetsregisteret/api/enheter/{orgNumber}";
+            _logger.LogInformation($"Retrieving organization {orgNumber} from BRREG, url: {_client.BaseAddress + path}");
+
+            var response = await _client.GetAsync(path);
+
+            if (response.StatusCode != HttpStatusCode.OK)
+            {
+                _logger.LogWarning($"Failed to retrieve organization {orgNumber}, status code: {response.StatusCode}");
+                return null;
+            }
+
+            var data = await response.Content.ReadAsStringAsync();
+            var brregOrg = JsonSerializer.Deserialize<BrregOrg>(data, _serializerOptions);
+
+            if (brregOrg == null) return brregOrg;
```

Comment:
Burde vel logge noe her også... type "Failed to parse organization for {orgNumber}" elns